cmake/modules/AddSwift.cmake

```diff
@@ -289,10 +295,11 @@ function(_add_variant_c_compile_flags)
   if("${CFLAGS_SDK}" STREQUAL "ANDROID")
     # FIXME: Instead of hardcoding paths in the Android NDK, these paths should
     #        be passed in via ENV, as with the Windows build.
-    list(APPEND result
-        "-I${SWIFT_ANDROID_NDK_PATH}/sources/cxx-stl/llvm-libc++/include"
-        "-I${SWIFT_ANDROID_NDK_PATH}/sources/cxx-stl/llvm-libc++abi/include"
-        "-I${SWIFT_ANDROID_NDK_PATH}/sources/android/support/include")
+    swift_android_include_for_arch("${CFLAGS_ARCH}" "${CFLAGS_ARCH}_INCLUDE")
+    foreach(path IN LISTS ${CFLAGS_ARCH}_INCLUDE)
+      list(APPEND result "-I${path}")
+    endforeach()
```

Comment:
I think that we should introduce a `swift_include_directories` similar to `swift_target_link_search_directories`.  Use `CMAKE_INCLUDE_FLAG_C` rather than `-I` as `cl` uses `/I`.